### ISSUE TITLE
jsk ast fails with an error in certain cases

### ISSUE THREAD

**Issue by pythonmcpi** [Gorialis/jishaku#190]
## Summary

Running the following produces an error:

&jsk ast 
```python
def h(*, a):
    print(a)
```

Traceback:
```python
Traceback (most recent call last):
  File "C:\Users\**\AppData\Local\Programs\Python\Python310\lib\site-packages\jishaku\features\python.py", line 413, in jsk_ast
    text = create_tree(argument.content, use_ansi=Flags.use_ansi(ctx))
  File "C:\Users\**\AppData\Local\Programs\Python\Python310\lib\site-packages\jishaku\repl\disassembly.py", line 202, in create_tree
    return '\n'.join(format_ast_node(user_code, use_ansi=use_ansi))
  File "C:\Users\**\AppData\Local\Programs\Python\Python310\lib\site-packages\jishaku\repl\disassembly.py", line 187, in format_ast_node
    yield from format_ast_block(
  File "C:\Users\**\AppData\Local\Programs\Python\Python310\lib\site-packages\jishaku\repl\disassembly.py", line 163, in format_ast_block
    for child_index, description in enumerate(format_ast_node(item, level=level + 1, use_ansi=use_ansi)):
  File "C:\Users\**\AppData\Local\Programs\Python\Python310\lib\site-packages\jishaku\repl\disassembly.py", line 187, in format_ast_node
    yield from format_ast_block(
  File "C:\Users\**\AppData\Local\Programs\Python\Python310\lib\site-packages\jishaku\repl\disassembly.py", line 163, in format_ast_block
    for child_index, description in enumerate(format_ast_node(item, level=level + 1, use_ansi=use_ansi)):
  File "C:\Users\**\AppData\Local\Programs\Python\Python310\lib\site-packages\jishaku\repl\disassembly.py", line 187, in format_ast_node
    yield from format_ast_block(
  File "C:\Users\**\AppData\Local\Programs\Python\Python310\lib\site-packages\jishaku\repl\disassembly.py", line 163, in format_ast_block
    for child_index, description in enumerate(format_ast_node(item, level=level + 1, use_ansi=use_ansi)):
AttributeError: 'NoneType' object has no attribute '_fields'
```

## Reproduction steps

Just run the snippet above. You can also try it with any function that has a keyword-only argument without a default value.

## Expected results

Ast tree prints out correctly.

## Actual results

A traceback is sent to me by the bot and no ast is sent.

## Checklist

- [x] I have updated discord.py and jishaku to the latest available versions and have confirmed that this issue is still present
- [x] I have searched the open issues for duplicates
- [x] I have shown the entire traceback, if possible
- [x] I have removed my token from display, if visible

## System information

```
$ py -3.10 -m discord -v
- Python v3.10.4-final
- discord.py v2.1.0-final
- aiohttp v3.8.1
- system info: Windows 10 10.0.19044

$ py -3.10 -m pip show jishaku
Name: jishaku
Version: 2.5.1
Summary: A discord.py extension including useful tools for bot development and debugging.
Home-page: https://github.com/Gorialis/jishaku
Author: Devon (Gorialis) R
Author-email:
License: MIT
Location: c:\users\**\appdata\local\programs\python\python310\lib\site-packages
Requires: braceexpand, click, import-expression, line-profiler
Required-by:
```

## Further Notes

This bug was discovered while trying to debug some bot code I wrote in [hy](https://github.com/hylang/hy). I had written a custom `hy_ast` command that parsed the ast tree and sent the tree to `jishaku.repl.disassembly.format_ast_node`. I initally thought it was an issue with my code, but I was able to replicate the bug with some python source.

The hy version the erroring code above is `(defn h [* a] (print a))`. Running `ast.unparse(...)` on the ast tree produced by hy gives the exact same python code as the example in the reproduction steps.

The following is the output of `ast.dump(ast.parse(code), indent=4)` for two snippets:

```python
def h(*, a):
    print(a)
```
gives
```python
Module(
    body=[
        FunctionDef(
            name='h',
            args=arguments(
                posonlyargs=[],
                args=[],
                kwonlyargs=[
                    arg(arg='a')],
                kw_defaults=[
                    None],
                defaults=[]),
            body=[
                Expr(
                    value=Call(
                        func=Name(id='print', ctx=Load()),
                        args=[
                            Name(id='a', ctx=Load())],
                        keywords=[]))],
            decorator_list=[])],
    type_ignores=[])
```

and
```python
def h(*, a=None):
    print(a)
```
gives
```python
Module(
    body=[
        FunctionDef(
            name='h',
            args=arguments(
                posonlyargs=[],
                args=[],
                kwonlyargs=[
                    arg(arg='a')],
                kw_defaults=[
                    Constant(value=None)],
                defaults=[]),
            body=[
                Expr(
                    value=Call(
                        func=Name(id='print', ctx=Load()),
                        args=[
                            Name(id='a', ctx=Load())],
                        keywords=[]))],
            decorator_list=[])],
    type_ignores=[])
```

It seems like the only difference is in the kw_defaults list on the FunctionDef args attribute.

**Comment by AbstractUmbra** [comment]
I can also reproduce this:

<details open>
<summary>Traceback</summary>
<br>

```py
Traceback (most recent call last):
  File "/home/penumbra/.cache/pypoetry/virtualenvs/mipha-kl4lG4Iw-py3.10/lib/python3.10/site-packages/jishaku/features/python.py", line 414, in jsk_ast
    text = create_tree(argument.content, use_ansi=Flags.use_ansi(ctx))
  File "/home/penumbra/.cache/pypoetry/virtualenvs/mipha-kl4lG4Iw-py3.10/lib/python3.10/site-packages/jishaku/repl/disassembly.py", line 202, in create_tree
    return '\n'.join(format_ast_node(user_code, use_ansi=use_ansi))
  File "/home/penumbra/.cache/pypoetry/virtualenvs/mipha-kl4lG4Iw-py3.10/lib/python3.10/site-packages/jishaku/repl/disassembly.py", line 187, in format_ast_node
    yield from format_ast_block(
  File "/home/penumbra/.cache/pypoetry/virtualenvs/mipha-kl4lG4Iw-py3.10/lib/python3.10/site-packages/jishaku/repl/disassembly.py", line 163, in format_ast_block
    for child_index, description in enumerate(format_ast_node(item, level=level + 1, use_ansi=use_ansi)):
  File "/home/penumbra/.cache/pypoetry/virtualenvs/mipha-kl4lG4Iw-py3.10/lib/python3.10/site-packages/jishaku/repl/disassembly.py", line 187, in format_ast_node
    yield from format_ast_block(
  File "/home/penumbra/.cache/pypoetry/virtualenvs/mipha-kl4lG4Iw-py3.10/lib/python3.10/site-packages/jishaku/repl/disassembly.py", line 163, in format_ast_block
    for child_index, description in enumerate(format_ast_node(item, level=level + 1, use_ansi=use_ansi)):
  File "/home/penumbra/.cache/pypoetry/virtualenvs/mipha-kl4lG4Iw-py3.10/lib/python3.10/site-packages/jishaku/repl/disassembly.py", line 187, in format_ast_node
    yield from format_ast_block(
  File "/home/penumbra/.cache/pypoetry/virtualenvs/mipha-kl4lG4Iw-py3.10/lib/python3.10/site-packages/jishaku/repl/disassembly.py", line 163, in format_ast_block
    for child_index, description in enumerate(format_ast_node(item, level=level + 1, use_ansi=use_ansi)):
AttributeError: 'NoneType' object has no attribute '_fields'
```

</br>
</details>

System details are:
```sh
$ python -m discord -v    
- Python v3.10.8-final
- discord.py v2.2.0-alpha
    - discord.py metadata: v2.2.0
- aiohttp v3.8.3
- system info: Linux 5.10.0-15-amd64 #1 SMP Debian 5.10.120-1 (2022-06-09)

$ pip show jishaku                               
Name: jishaku
Version: 2.5.1a460+g7777777.master
Summary: A discord.py extension including useful tools for bot development and debugging.
Home-page: https://github.com/Gorialis/jishaku
Author: Devon (Gorialis) R
Author-email: 
License: MIT
Location: /home/penumbra/.cache/pypoetry/virtualenvs/mipha-kl4lG4Iw-py3.10/lib/python3.10/site-packages
Requires: braceexpand, click, import-expression, tabulate
Required-by: 
```

**Comment by Gorialis** [comment]
Fixed in 167b9ca281e7ddbf2f9bf2a1784c522967259b9f
